Adding Portuguese localization

milliet opened the pull request:
Files created and modified to add "pt" (Portuguese) to languages available.

horus68 commented:
Good effort, you still have many errors to correct in the language, or is this a Portuguese (Brasil) translation? 
If so the language code needs to be different to have different files for Portugal and Brasil !

Note: I've been waiting that my proposal to create a Crowdin or Transifex system is implemented!

milliet commented:
Yes, this is a Brazilian Portuguese language. The translation is already
finished. We have built a working Mac version, but still couldn't make a
windows one.

Anyway, you're right about having a pt and a pt-br versions in the Lang
catalog. We'll do it, bit it seems we have to change all catalogs.

The European portuguese would need minor adjustments, but it is almost the
same.

Thanks

Pedro

Em sáb, 8 de ago de 2020 15:15, Horus68 <notifications@github.com> escreveu:

> Good effort, you still have many errors to correct in the language, or is
> this a Portuguese (Brasil) translation?
> If so the language code needs to be different to have different files for
> Portugal and Brasil !
>
> Note: I've been waiting that my proposal to create a Crowdin or Transifex
> system is implemented!
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/edrlab/thorium-reader/pull/1141#issuecomment-670958124>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AALE6OQTB6V4QMMIVCH5S5TR7WI47ANCNFSM4PYBMCSQ>
> .
>

horus68 commented:
Thanks. If you can please duplicate the translations to have a pt-PT and a pt-BR version for this PR.
I will then update the pt-PT version later, when merged.
@danielweck requesting your yes on this!

horus68 commented on src/resources/locales/pt.json:
```diff
@@ -0,0 +1,336 @@
+            "editUserJson": "Editar arquvo de usuário...",
+            "hide": "Ocultar",
+            "keyboardShortcuts": "Atalhos de teclado",
+            "loadUserJson": "Carregar de arquivo de usuário",
```
typo: change arquvo to arquivo

horus68 commented on src/resources/locales/pt.json:
```diff
@@ -0,0 +1,336 @@
+            "next": "Próxima narração",
+            "pause": "Pausar narração",
+            "play": "Tocar narração",
+            "previous": "Naração anterior",
```
Typo: change "Naração" to "Narração"

horus68 commented on src/resources/locales/pt.json:
```diff
@@ -0,0 +1,336 @@
+            "error": "Incapaz de acessar a origem OPDS",
+            "noInternet": "Sem Internet",
+            "noInternetMessage": "Tente: checar o cabo de rede, modem, router ou wifi",
+            "reject": "Desculpe, houve um erro inesperado tentanto processar a origem OPDS",
```
Tyo: change "tentanto" to "tentando"

horus68 commented on src/resources/locales/pt.json:
```diff
@@ -0,0 +1,336 @@
+        },
+        "publisher": "Editora",
+        "readBook": "Ler",
+        "released": "Pubolicado em",
```
Typo: change "Pubolicado" to "Pubilicado"

milliet commented:
Thank you @horus68! I'll make the corrections  you've pointed out and include pt-pt as you suggest.

danielweck commented:
Thank you Pedro and "Horus68", much appreciated :)

milliet commented:
Hi @horus68! I hope you're fine there. I've fixed the typos and created files for pt-pt.json and pt-br.json, pt-pt.md and pt-br.md (in src/resources/information), updated ptLang.json (in src/resources/locale-names) to include both versions for Portuguese (Portugal and Brasileiro). I didn't change /src/common/services/translator.ts because I didn't know what to do with it, but it seems it must be changed to allow four letter languages. 
Please, tell me if I need to do any other change.

Hi @danielweck. Nice to be here! Thanks for helping.

horus68 commented:
@milliet Thanks for your work.
Hope this PR gets merged soon. Then I can work to improve the pt-PT files.
@danielweck It would also require a dev to implement the four letter code languages. This should be a system wide config as there are many other languages requiring this!

danielweck commented on src/common/services/translator.ts:
```diff
@@ -13,13 +13,15 @@ import * as esCatalog from "readium-desktop/resources/locales/es.json";
 import * as frCatalog from "readium-desktop/resources/locales/fr.json";
 import * as jaCatalog from "readium-desktop/resources/locales/ja.json";
 import * as nlCatalog from "readium-desktop/resources/locales/nl.json";
+import * as ptCatalog from "readium-desktop/resources/locales/pt.json";
```
This should be:

```
import * as ptBrCatalog from "readium-desktop/resources/locales/pt-br.json";
import * as ptPtCatalog from "readium-desktop/resources/locales/pt-pt.json";
```

danielweck commented on src/common/services/translator.ts:
```diff
@@ -95,6 +97,9 @@ i18nextInstance.init({
         },
+        pt: {
+            translation: ptCatalog,
+        },
```
This should be:

```
        ptBr: {
            translation: ptBrCatalog,
        },
        ptPt: {
            translation: ptPtCatalog,
        },
```

danielweck replied:
Sorry, wrong key syntax, I meant:

```
       pt-br: {
            translation: ptBrCatalog,
        },
        pt-pt: {
            translation: ptPtCatalog,
        },
```

danielweck commented on src/common/services/translator.ts:
```diff
@@ -122,6 +127,7 @@ i18nextInstance.addResourceBundle("de", "translation", deLang, true);
 i18nextInstance.addResourceBundle("es", "translation", esLang, true);
 i18nextInstance.addResourceBundle("nl", "translation", nlLang, true);
 i18nextInstance.addResourceBundle("ja", "translation", jaLang, true);
+i18nextInstance.addResourceBundle("pt", "translation", ptLang, true);
```
This should be:

```
i18nextInstance.addResourceBundle("pt-br", "translation", ptBrLang, true);
i18nextInstance.addResourceBundle("pt-pt", "translation", ptPtLang, true);
```

danielweck commented on src/common/services/translator.ts:
```diff
@@ -140,6 +146,7 @@ export const AvailableLanguages = {
     es: "Español",
     nl: "Dutch",
     ja: "日本語",
+    pt: "Português",
```
This should be:

```
"pt-br": "Português Brasileiro",
"pt-pt": "Português",
```

danielweck commented:
Thank you very much! I added comments to the PR source code.

horus68 commented:
@milliet Pedro... your turn now!

milliet commented:
@danielweck, translator.ts Done! Tell me if you need any other action. Thanks.
@horus68 , it seems it is fine for now, but tell me about anything needed.

horus68 commented:
@milliet ptLang.jso should also be duplicated with the proper name code change

horus68 commented on src/common/services/translator.ts:
```diff
@@ -13,13 +13,16 @@ import * as esCatalog from "readium-desktop/resources/locales/es.json";
 import * as frLang from "readium-desktop/resources/locale-names/frLang.json";
 import * as jaLang from "readium-desktop/resources/locale-names/jaLang.json";
 import * as nlLang from "readium-desktop/resources/locale-names/nlLang.json";
+import * as ptLang from "readium-desktop/resources/locale-names/ptLang.json";
```
The file ptLang.json needs to be duplicated for pt-pt and pt-br

milliet replied:
Changes made as suggested, 1 file created (pt-brLang.json) and one renamed (pt-ptLang.json). translator.ts changed to replace ptLang.json by both files. Thanks and waiting for any other change needed.

milliet commented:
@horus68 and @danielweck , it seems it's OK now. I had to change all lang codes to quoted in translator.ts (it didn't pass Mac and Linux before that).

danielweck commented:
@horus68 could you please “resolve conversation” (open comments) wherever appropriate in the discussion thread? Once confirmed, we will merge the PR.

horus68 commented:
@danielweck I think I approved it all now (this github feature its not very intuitive)
